src/index.c

```diff
@@ -527,6 +529,7 @@ int git_index_clear(git_index *index)
 
 	assert(index);
 
+	index->dirty = 1;
```

Comment:
Unsure, but this seems the inverse of the commit's message rationale ?

Reply:
Well. Clearing the index here without saving naturally has to leave the index dirty. I agree though that the commit message is confusing